apps/web/pages/[...slug].vue

```diff
@@ -50,24 +49,28 @@ const handleQueryUpdate = async () => {
   }
 };
 
-await handleQueryUpdate();
-
-const category = ref(productsCatalog.value.category);
+onNuxtReady(async () => {
+  await handleQueryUpdate().then(() => setCategoriesPageMeta(productsCatalog.value, getFacetsFromURL()));
+});
 
 const breadcrumbs = computed(() => {
-  const breadcrumb = categoryTreeGetters.generateBreadcrumbFromCategory(
-    categoryTree.value,
-    categoryGetters.getId(category.value),
-  );
-  breadcrumb.unshift({ name: t('home'), link: '/' });
+  if (productsCatalog.value.category) {
+    const breadcrumb = categoryTreeGetters.generateBreadcrumbFromCategory(
+      categoryTree.value,
+      categoryGetters.getId(productsCatalog.value.category),
+    );
+    breadcrumb.unshift({ name: t('home'), link: '/' });
 
-  return breadcrumb;
+    return breadcrumb;
+  }
+
+  return [];
 });
 
 watch(
   () => locale.value,
   async (changedLocale: any) => {
-    router.push({
+    await router.push({
```

Comment:
why do we need the await here. Also from what I know a watch that is async has to be saved into a variable and the watch has to be destroyed when the component unmounts. https://vuejs.org/guide/essentials/watchers.html#stopping-a-watcher

Reply:
ok if thats the case then there is nothing to worry about